public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -582,3 +589,38 @@ function validateRowsLayout(layout: unknown) {
     throw new Error('Layout spec items is not an array');
   }
 }
+
+/**
+ * Get a collection of panel queries refIds
+ * the refIds are the ones which did not have a datasource set
+ * @returns a set of panel queries refIds
+ */
+function getPanelDsReferences(vizPanel: VizPanel) {
```

Comment:
Nit: I find the name confusing or abstract in general. Panel DS references, in the end, are panel DS that were undefined initially, which means that they were defaulted in runtime and should be stored as default DS, aka undefined.

Something like `getPanelDsReferences` could be called something like `getDefaultedPanelQueries` `getAutoAssignedPanelDS`, `getPanelDefaultedDS` or even `getImplicitPanelDataSources`.

WDYT? :D